src/main/java/seedu/address/logic/commands/UnlabelCommand.java

```diff
@@ -0,0 +1,115 @@
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes Label(s) in an existing Tag. "
+            + "\nParameters: "
+            + PREFIX_TAG_NAME + "TAG_NAME "
+            + PREFIX_LABEL_NAME + "LABEL_NAME "
+            + "[" + PREFIX_LABEL_NAME + "LABEL_NAME" + "]...\n"
+            + "\nExample: " + COMMAND_WORD + " "
+            + PREFIX_TAG_NAME + "cs2103 "
+            + PREFIX_LABEL_NAME + "notes "
+            + PREFIX_LABEL_NAME + "CS2103T";
+
+    public static final String MESSAGE_SUCCESS = "Tag successfully updated: %1$s";
+    public static final String MESSAGE_LABEL_MISSING = "All of this label can't be found in %s:\n";
+    public static final String MESSAGE_DASH = "- %s\n";
+    public static final String MESSAGE_INVALID_LABEL = "The remaining labels that are not "
+            + "listed above have been deleted.";
+    public static final String MESSAGE_TAG_NOT_FOUND = "Tag: %s not found!"
+            + " Please make sure that Tag is present before deleting its label.";
+
+    private final TagName tagName;
+    private final Set<Label> labels;
+
+    /**
+     * Creates an UnlabelCommand to delete labels to the specified {@code tagName}
+     */
+    public UnlabelCommand(TagName tagName, Set<Label> labels) {
+        this.tagName = tagName;
+        this.labels = labels;
+    }
+
+    /**
+     * Executes the command and deletes the specified tag's labels according to the arguments.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return A Command result of executing unlabel command.
+     * @throws CommandException
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
+        // Check if tag exists
+        List<Tag> tagList = model.findFilteredTagList(new TagNameEqualsKeywordPredicate(tagName));
+        if (tagList.isEmpty()) {
+            throw new CommandException(String.format(MESSAGE_TAG_NOT_FOUND, tagName.toString()));
+        }
+
+        // Get the tag
+        Tag tagToChange = tagList.get(0);
+        FileAddress fileAddress = tagToChange.getFileAddress();
+
+        // Deletes the current tag from the model
+        model.deleteTag(tagToChange);
+
+        // Deletes all the matching labels
+        Set<Label> editedLabel = tagToChange.getLabels().stream()
+                .filter(label -> {
+                    if (labels.stream().anyMatch(label::equals)) {
+                        labels.remove(label);
+                        return false;
+                    }
+
+                    return true;
+                })
+                .collect(Collectors.toSet());
+
+        Tag editedTag = new Tag(tagName, fileAddress, editedLabel);
+
+        // Deletes the edited tag from the model
+        model.addTag(editedTag);
+
+        model.commitAddressBook();
+
+        if (labels.isEmpty()) {
+            return new CommandResult(String.format(MESSAGE_SUCCESS, editedTag));
+        }
```

Comment:
Happy path is not prominent here. Maybe you should change this to
```
if (!labels.isEmpty()) {
...
}

return new CommandResult(String.format(MESSAGE_SUCCESS, editedTag));
```

Reply:
Noted!